pkg/admission/zpdb_validator.go

```diff
@@ -0,0 +1,97 @@
+package admission
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/spanlogger"
+	v1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/grafana/rollout-operator/pkg/zpdb"
+)
+
+const (
+	ZpdbValidatorWebhookPath = "/admission/zpdb-validation"
+)
+
+type zpdbValidatingHandler struct {
+	ctx     context.Context
+	logger  *spanlogger.SpanLogger
+	request v1.AdmissionReview
+}
+
+func (v *zpdbValidatingHandler) initLogger() {
+	v.logger.SetSpanAndLogTag("object.name", v.request.Request.Name)
+	v.logger.SetSpanAndLogTag("object.resource", v.request.Request.Resource.Resource)
+	v.logger.SetSpanAndLogTag("object.namespace", v.request.Request.Namespace)
+	// note that this is the UID of the request, not of the pod
+	v.logger.SetSpanAndLogTag("request.uid", v.request.Request.UID)
+
+	if v.request.Request.DryRun != nil {
+		v.logger.SetSpanAndLogTag("request.dry_run", v.request.Request.DryRun)
+	}
+}
+
+// parse attempts to parse the raw object to a ZpdbConfig (ZoneAwarePodDisruptionBudget).
+// returns an error and http status code if the parse or validation fails
+func (v *zpdbValidatingHandler) parse() (int32, error) {
+	var obj unstructured.Unstructured
+	if err := json.Unmarshal(v.request.Request.Object.Raw, &obj); err != nil {
+		level.Info(v.logger).Log(logMsg, errors.New("failed to unmarshal object"), "err", err)
+		return int32(404), err
```

Comment:
Does 404 (not found) make sense in this context? 400 (bad request) feels like a better fit to me.

It'd also be good to use the constants from the `net/http` package here (eg. `http.StatusBadRequest`).